### PR TITLE
Adds install.sh script to simplify download of pre-built release  binaries

### PR DESCRIPTION
**Pull request by sameersbn**

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -24,6 +24,9 @@ builds:
 archives:
   - id: default
     format: binary
+    name_template: '{{ .Binary }}_{{ .Os }}_{{ .Arch }}{{ with .Arm }}v{{ . }}{{ end }}{{ with .Mips }}_{{ . }}{{ end }}{{ if not (eq .Amd64 "v1") }}{{ .Amd64 }}{{ end }}'
+    replacements:
+      darwin: macOS
```
Comment (sameersbn) [inline .goreleaser.yaml:29]: macOS binaries will have the string "macOS" as opposed to "darwin"

**Comment by tzununbekov** [comment]
As I said, I'm paranoid about blind redirects of internet scripts to my Sudo'ed shell. If you think it's ok, then it's fine with me.

**Comment by sameersbn** [comment]
> As I said, I'm paranoid about blind redirects of internet scripts to my Sudo'ed shell. If you think it's ok, then it's fine with me.

It's just a convenience wrapper, user's can go to the releases page and download the binary if needed. Will leave it up to   @sebgoa @jmcx to take the call.

**Comment by sebgoa** [comment]
I am also paranoid and others too, but we should have it and give a warning on the docs.

@jmcx something like "If you trust us :) feel free to use this script !"

File: README.md [hunk 2]
```diff
@@ -21,13 +21,10 @@ brew install tmctl
...
-Linux, MacOS:
+Linux, macOS and Windows
 
-```
-export TMCTL_VERSION=$(curl -s -I HEAD https://github.com/triggermesh/tmctl/releases/latest | grep "location:" | awk -F / '{print $NF}')
-curl -L https://github.com/triggermesh/tmctl/releases/download/$TMCTL_VERSION/tmctl_${TMCTL_VERSION:1}_$(uname -m -o | awk '{print tolower($1)"_"$2}') -o tmctl \
-    && chmod +x tmctl \
-    && sudo mv tmctl /usr/local/bin
+```bash
+curl -sSfL https://raw.githubusercontent.com/triggermesh/tmctl/HEAD/scripts/install.sh | sh
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply README.md:35]: @sebgoa rewrote this section, PTAL

**Comment by tzununbekov** [comment]
Maybe we could move this script to the `hack/` directory where we [usually](https://github.com/triggermesh/triggermesh/tree/main/hack) store all scripts? If it doesn't make the installation command more suspicious of course 😅

**Comment by sameersbn** [comment]
> Maybe we could move this script to the `hack/` directory where we [usually](https://github.com/triggermesh/triggermesh/tree/main/hack) store all scripts? If it doesn't make the installation command more suspicious of course 😅

@tzununbekov i've seen projects create a separate repo (fe. homebrew/install) for such scripts. Along similar lines, we could create `triggermesh/tmctl-install` with the `install.sh` script to start with, later we can add `uninstall.sh` or other such scripts. what do you think?

**Comment by sebgoa** [comment]
no real opinion here. `/hack` has been k8s and knative go to place to put these things I think.

**Comment by sameersbn** [comment]
in the interest of not wasting too much time, i'm going to move it to the `hack/` directory

**Comment by tzununbekov** [comment]
> > Maybe we could move this script to the `hack/` directory where we [usually](https://github.com/triggermesh/triggermesh/tree/main/hack) store all scripts? If it doesn't make the installation command more suspicious of course 😅
> 
> @tzununbekov i've seen projects create a separate repo (fe. homebrew/install) for such scripts. Along similar lines, we could create `triggermesh/tmctl-install` with the `install.sh` script to start with, later we can add `uninstall.sh` or other such scripts. what do you think?

not quite sure what purpose will separate repo have, maybe it is something specific to the brew release approach. Right now having everything related in the same place feels more convenient. I'd vote for `hack/` directory, but it's not mandatory. I'll approve the PR with the assumption that the script is working.

**Comment by sameersbn** [comment]
updated.. please review

**Review by tzununbekov — APPROVED** [review]

**Comment by tzununbekov** [comment]
Could you also hold off the test release for ~30 mins, I have another PR coming that I'd want to be in before

**Pull request merged**